content/docs/getting-started/hello-world.md

```diff
@@ -71,93 +56,74 @@ Next, we'll add some to the `client` `TcpStream`. This asynchronous task now cre
 the stream and then yields it once it's been created for additional processing.
 
 ```rust
-# #![deny(deprecated)]
-# extern crate tokio;
-#
-# use tokio::net::TcpStream;
-# use tokio::prelude::*;
-# fn main() {
-# let addr = "127.0.0.1:6142".parse().unwrap();
-let client = TcpStream::connect(&addr).and_then(|stream| {
-    println!("created stream");
-
-    // Process stream here.
-
-    Ok(())
-})
-.map_err(|err| {
-    // All tasks must have an `Error` type of `()`. This forces error
-    // handling and helps avoid silencing failures.
-    //
-    // In our example, we are only going to log the error to STDOUT.
-    println!("connection error = {:?}", err);
-});
-# }
+use tokio::net::TcpStream;
+
+fn main() {
+    let addr = "127.0.0.1:6142";
+    let client = TcpStream::connect(addr);
+
+    let future = async move {
+        match client.await {
+            Ok(_stream) => {
+                println!("created stream");
+
+                // Process stream here.
+            }
+            Err(err) => {
+                // In our example, we are only going to log the error to STDOUT.
+                println!("connection error = {:?}", err);
+            }
+        };
```

Comment:
We might want to consider modifying this code to just use the `?` operator — removing error handling boilerplate would let users focus on the new content being introduced. Maybe something like:
```rust
let future = async move {
    let stream = client.await?;
    println!("created stream");
# drop(stream); // this lets the `stream` local appear to be used so we don't need `_stream`

    // process stream here

    Ok(())
};
```
and add a comment explaining that we can use `?` in an `async` block the way we can in other code.